velox/exec/tests/AggregationFuzzerTest.cpp

```diff
@@ -257,6 +259,65 @@ getCustomInputGenerators() {
   };
 }
 
+/// Applies specified SQL transformation to the results before comparing. For
+/// example, sorts an array before comparing results of array_agg.
+///
+/// Supports 'compare' API.
+class TransformResultVerifier : public ResultVerifier {
+ public:
+  /// @param transform fmt::format-compatible SQL expression to use to transform
+  /// aggregation results before comparison. The string must have a single
+  /// placeholder for the column name that contains aggregation results. For
+  /// example, "array_sort({})".
+  TransformResultVerifier(const std::string& transform)
+      : transform_{transform} {}
+
+  static std::shared_ptr<ResultVerifier> create(const std::string& transform) {
+    return std::make_shared<TransformResultVerifier>(transform);
+  }
+
+  bool supportsCompare() override {
```

Comment:
nit: I wonder if we can pass in comparison or verification as a lambda so we might not need the supports functions ?

Reply:
These 2 API are different and we need a way to know which API is supported.